dotnet/Vaas/src/Vaas/Messages/AuthenticationResponse.cs

```diff
@@ -0,0 +1,19 @@
+using System.Text.Json.Serialization;
+
+namespace Vaas.Messages
+{
+    public class AuthenticationResponse
+    {
+        [JsonPropertyName("kind")] 
+        public string Kind = "AuthResponse";
+        
+        [JsonPropertyName("success")] 
+        public bool Success { get; init; }
+
+        [JsonPropertyName("session_id")] 
+        public string SessionId { get; init; } = null!;
```

Comment:
That strange syntax come from the .NET deserializer implementation. The "*Response" types are never instantiated in code but only from the deserializer.

The value must not be null and has to throw an exception if a null value gets deserialized. On the other hand, it needs to be "init", such that that the deserializer can write to it, but it is immutable after. That's why "set" or only "get" won't work.

Setting it to the empty string won't work, as a missing field would be deserialized to an empty string, instead of an exception.

Not happy with that code,  but the only why Anni and me came up with a way to encode the requirements correctly in code. Not the strongest type system here...

I'm open to any more elegant solutions :)

Reply:
Auf diese Weise hast du leider "worst-of-both-worlds": Zwar kompiliert es, aber er schmeißt keine Exception, wenn die Property nicht im JSON existiert und der Wert kann nach der Deserialisierung null sein, obwohl der Typ nicht nullable ist. S. diesen Test. Weil ich mir auch nicht sicher war, habe ich das gerade getestet.

```C#
public class Model
{
    public string A { get; set; } = null!;
}

public class SerializerTest
{
    [Fact]
    public void ThrowsException()
    {
        string json = "{}";
        var model = JsonSerializer.Deserialize<Model>(json);
        Assert.Equal(null, model.A);
    }
}
```

Bei Json.net konnte man [Required] setzen. System.Text.Json hat noch keine Implementierung, aber eventuell hilft das https://github.com/dahomey-technologies/Dahomey.Json . Die Lib stellt Attribute zu Verfügung. Klappt auch.

Reply:
It's a bit better now. For the future: Use proper model validation when System.Text.Json supports it.